ksql-rest-app/src/test/java/io/confluent/ksql/api/BasicAuthTest.java

```diff
@@ -0,0 +1,230 @@
+        ApiServerConfig.AUTHENTICATION_METHOD_BASIC);
+    origs.put(
+        ApiServerConfig.AUTHENTICATION_REALM_CONFIG,
+        PROPS_JAAS_REALM
+    );
+    origs.put(
+        ApiServerConfig.AUTHENTICATION_ROLES_CONFIG,
+        KSQL_RESOURCE
+    );
+    return new ApiServerConfig(origs);
+  }
+
+  @Override
+  protected HttpResponse<Buffer> sendRequest(final WebClient client, final String uri,
+      final Buffer requestBody)
+      throws Exception {
+    return sendRequestWithCreds(client, uri, requestBody, USER_WITH_ACCESS, USER_WITH_ACCESS_PWD);
+  }
+
+  @Override
+  protected void streamRequest(
+      final WebClient client,
+      final String uri,
+      final Consumer<HttpRequest<Buffer>> requestSender) {
+    requestSender.accept(
+        client.post(uri)
+            .basicAuthentication(USER_WITH_ACCESS, USER_WITH_ACCESS_PWD)
+    );
+  }
+
+  @Test
+  public void shouldFailPullQueryWithBadCredentials() throws Exception {
+    shouldFailPullQuery(USER_WITHOUT_ACCESS, USER_WITHOUT_ACCESS_PWD);
+  }
+
+  @Test
+  public void shouldFailPushQueryWithBadCredentials() throws Exception {
+    shouldFailPushQuery(USER_WITHOUT_ACCESS, USER_WITHOUT_ACCESS_PWD);
+  }
+
+  @Test
+  public void shouldFailCloseQueryWithBadCredentials() throws Exception {
+    shouldFailCloseQuery(USER_WITHOUT_ACCESS, USER_WITHOUT_ACCESS_PWD);
+  }
+
+  @Test
+  public void shouldFailInsertRequestWithBadCredentials() throws Exception {
+    shouldFailInsertRequest(USER_WITHOUT_ACCESS, USER_WITHOUT_ACCESS_PWD);
+  }
+
+  @Test
+  public void shouldFailPullQueryWithIncorrectRole() throws Exception {
+    shouldFailPullQuery(USER_WITH_INCORRECT_ROLE, USER_WITH_INCORRECT_ROLE_PWD);
+  }
+
+  @Test
+  public void shouldFailPushQueryWithIncorrectRole() throws Exception {
+    shouldFailPushQuery(USER_WITH_INCORRECT_ROLE, USER_WITH_INCORRECT_ROLE_PWD);
+  }
+
+  @Test
+  public void shouldFailCloseQueryWithIncorrectRole() throws Exception {
+    shouldFailCloseQuery(USER_WITH_INCORRECT_ROLE, USER_WITH_INCORRECT_ROLE_PWD);
+  }
+
+  @Test
+  public void shouldFailInsertRequestWithIncorrectRole() throws Exception {
+    shouldFailInsertRequest(USER_WITH_INCORRECT_ROLE, USER_WITH_INCORRECT_ROLE_PWD);
+  }
+
+  private void shouldFailPullQuery(final String username, final String password) throws Exception {
+    // Given
+    JsonObject requestBody = new JsonObject().put("sql", DEFAULT_PULL_QUERY);
+    JsonObject properties = new JsonObject().put("prop1", "val1").put("prop2", 23);
+    requestBody.put("properties", properties);
+
+    // When
+    HttpResponse<Buffer> response = sendRequestWithCreds(
+        "/query-stream",
+        requestBody.toBuffer(),
+        username,
+        password
+    );
+
+    // Then
+    assertThat(response.statusCode(), is(401));
+    assertThat(response.statusMessage(), is("Unauthorized"));
+  }
+
+  private void shouldFailPushQuery(final String username, final String password) throws Exception {
```

Comment:
We probably don't need different tests for pull and push queries as they're handled on the same endpoint.

Reply:
Done.